odoo/addons/base/tests/test_ir_mail_server_smtpd.py

```diff
@@ -0,0 +1,651 @@
+        ))
+
+    def test_none_connection_starttls(self):
+        ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+        ssl_context.load_cert_chain(self.ssl_server.cert, self.ssl_server.key)
+        self.start_smtpd('starttls', ssl_context)
+        self.mail_server.write({
+            'smtp_encryption': 'none',
+            'smtp_authentication': 'login',
+            'smtp_user': '',
+            'smtp_pass': '',
+        })
+        with self.assertRaises(UserError) as capture:
+            self.mail_server.test_smtp_connection()
+        self.assertEqual(capture.exception.args[0], (
+            "The server refused the sender address (noreply@localhost) "
+            "with error b'Must issue a STARTTLS command first'"
+        ))
+
+    def test_none_password_invalid(self):
+        self.start_smtpd('none', auth_required=True)
+        self.mail_server.write({
+            'smtp_encryption': 'none',
+            'smtp_authentication': 'login',
+            'smtp_user': self.user_demo.email,
+            'smtp_pass': 'bad password',
+        })
+        with self.assertRaises(UserError) as capture:
+            self.mail_server.test_smtp_connection()
+        self.assertEqual(capture.exception.args[0], (
+            "The server has closed the connection unexpectedly. "
+            "Check configuration served on this port number.\n "
+            "Connection unexpectedly closed: timed out"
+        ))
+
+    def test_none_password_missing(self):
+        self.start_smtpd('none', auth_required=True)
+        self.mail_server.write({
+            'smtp_encryption': 'none',
+            'smtp_authentication': 'login',
+            'smtp_user': '',
+            'smtp_pass': '',
+        })
+        with self.assertRaises(UserError) as capture:
+            self.mail_server.test_smtp_connection()
+        self.assertEqual(capture.exception.args[0], (
+            "The server refused the sender address (noreply@localhost) "
+            "with error b'5.7.0 Authentication required'"
+        ))
+
+    def test_none_password_not_required(self):
+        self.start_smtpd('none', auth_required=False)
+        self.mail_server.write({
+            'smtp_encryption': 'none',
+            'smtp_authentication': 'login',
+            'smtp_user': '',
+            'smtp_pass': '',
+        })
+        self.mail_server.test_smtp_connection()
+
+    def test_none_password_valid(self):
+        self.start_smtpd('none', auth_required=True)
+        self.mail_server.write({
+            'smtp_encryption': 'none',
+            'smtp_authentication': 'login',
+            'smtp_user': self.user_demo.email,
+            'smtp_pass': PASSWORD,
+        })
+        self.mail_server.test_smtp_connection()
+
+    # -----------------------------------------------------
+    # smtp_encryption = 'ssl'
+    # -----------------------------------------------------
+    @unittest.skip("python doesn't validate certificate by default")
```

Comment:
shouldn't we not skip, and test that python doesn't check ?
(with a comment explaining that it's not a good thing, blabla, so if one day the test fail, we now it changed)